### PR TITLE
Refactoring Misc class.

### PR DESCRIPTION
**Pull request by tony-xia**
Thanks for your review.

### CODE REVIEW

File: src/Trinity.Core/Utilities/MISC/Misc.cs [hunk 1]
```diff
@@ -13,37 +13,28 @@ internal class Misc<T>
...
     internal class Misc
     {
         public static void DisplayByteArray(byte[] bytes)
         {
-            for (int i = 0; i < bytes.Length; )
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < bytes.Length; i++)
             {
-                for (int j = 0; j < 8; j++)
+                sb.Append(bytes[i++]);
```
Comment (yatli) [inline src/Trinity.Core/Utilities/MISC/Misc.cs]: with iterator incremented in the for loop it should not be pushed forward again here ;)

**Review by yatli — CHANGES_REQUESTED** [review]

File: src/Trinity.Core/Utilities/MISC/Misc.cs [hunk 2]
```diff
@@ -13,37 +13,28 @@ internal class Misc<T>
...
+                sb.Append('\t');
+
+                // 8 bytes per line
+                if (((i % 8) == 0) || (i == (bytes.Length - 1)))
```
Comment (leoxia) [inline src/Trinity.Core/Utilities/MISC/Misc.cs]: `if ((i+1) % 8 == 0 || i == bytes.Length - 1)` ?

**Review by leoxia — CHANGES_REQUESTED** [review]

**Review by leoxia — APPROVED** [review]
Thanks for the changes!

File: src/Trinity.Core/Utilities/MISC/Misc.cs (hunk 1 above)
Comment (tony-xia) [reply src/Trinity.Core/Utilities/MISC/Misc.cs]: You are correct.

File: src/Trinity.Core/Utilities/MISC/Misc.cs (hunk 2 above)
Comment (tony-xia) [reply src/Trinity.Core/Utilities/MISC/Misc.cs]: yes. thanks

**Review by yatli — APPROVED** [review]

**Pull request merged**